fe-library-app/src/components/Filter/Filter.tsx

```diff
@@ -1,5 +1,87 @@
-const Filter = () => {
-  return <div>Filter</div>
+import { ChangeEvent, useState } from 'react'
+
+import { Filters } from '../AppRouter/pages/HomePage'
+import styles from './Filter.module.css'
+
+interface FilterProps{
+  filterChange: (filterData: Filters) => void,
+  switchShowFilter: boolean,
+  hideFilter: () => void
+}
+
+const Filter = (props: FilterProps) => {
+  const initialState: Filters = {
```

Comment:
You can remove this const, there is no usage of this object in the code. As I see you always need one-by-one property.

You can do something like this instead:

`const [ desc, setDesc ] = useState('')
  const [ isbn, setIsbn ] = useState('')
  const [ authorFirstName, setAuthorFirstName ] = useState('')
  const [ authorLastName, setAuthorLastName ] = useState('')`